test/lib/parameters/validator_test.rb

```diff
@@ -0,0 +1,44 @@
+require 'test_helper'
+require "parameters/caster"
+
+class ValidatorTest < ActiveSupport::TestCase
+  class ValidatedItem
+    include ActiveModel::Validations
+    attr_accessor :value
+  end
+  before do
+    @item = ValidatedItem.new
+  end
+
+  it "adds errors on wrong regexp" do
+    @item.value = "123"
+    validator = Parameters::Validator.new(@item, :type => :regexp, :validate_with => "[a-z]", :getter => :value)
+    validator.validate!
```

Comment:
The return value of `validate!` is important in classification/base.rb, so I'd suggest assert/refuting it in all of these tests as well as checking errors.